compiler/passes/normalize.cpp

```diff
@@ -1952,18 +1952,66 @@ static void normVarTypeWoutInit(DefExpr* defExpr) {
     var->type = type;
 
   } else {
-    VarSymbol* typeTemp = newTemp("type_tmp");
-    DefExpr*   typeDefn = new DefExpr(typeTemp);
-    CallExpr*  initCall = new CallExpr(PRIM_INIT, typeExpr);
-    CallExpr*  initMove = new CallExpr(PRIM_MOVE, typeTemp, initCall);
+    bool handled = false;
```

Comment:
I think the "bool handled;" path is unwieldy.  As I understand it, handled will be true if
ts->type is a generic class or a generic record with initializers.  Why not make that a predicate
that can be invoked before dropping in to the else?